lib/net.js

```diff
@@ -1080,12 +1080,14 @@ function Server(options, connectionListener) {
     connectionListener = options;
     options = {};
     self.on('connection', connectionListener);
-  } else {
+  } else if(options == null || typeof options === 'object') {
```

Comment:
`typeof null` returns `'object'` so you can skip the check for `options == null` unless the intention is for the not-strict equality to allow `undefined`, in which case I imagine checking for `undefined` would be more clear.